src/matrixrtc/MatrixRTCSessonManager.ts

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { logger } from "../logger";
+import { MatrixClient, ClientEvent } from "../client";
+import { TypedEventEmitter } from "../models/typed-event-emitter";
+import { Room } from "../models/room";
+import { RoomState, RoomStateEvent } from "../models/room-state";
+import { MatrixEvent } from "../models/event";
+import { MatrixRTCSession } from "./MatrixRTCSession";
+
+export enum MatrixRTCSessionManagerEvents {
+    // A member has joined the MatrixRTC session, creating an active session in a room where there wasn't previously
+    SessionStarted = "session_started",
+    // All participants have left a given MatrixRTC session.
+    SessionEnded = "session_ended",
+}
+
+type EventHandlerMap = {
+    [MatrixRTCSessionManagerEvents.SessionStarted]: (roomId: string, session: MatrixRTCSession) => void;
+    [MatrixRTCSessionManagerEvents.SessionEnded]: (roomId: string, session: MatrixRTCSession) => void;
+};
+
+export class MatrixRTCSessionManager extends TypedEventEmitter<MatrixRTCSessionManagerEvents, EventHandlerMap> {
+    // All the room-scoped sessions we know about. This will include any where the app
+    // has queried for the MatrixRTC sessions in a room, whether it's ever had any members
+    // or not)
+    private roomSessions = new Map<string, MatrixRTCSession>();
+
+    public constructor(private client: MatrixClient) {
```

Comment:
Could we please use an `opts` object here in case the amount of params grows?

Reply:
Again I'd argue since this isn't part of the library interface, it doesn't really matter.